.github/workflows/build-and-test-python-wheels.yml

```
@@ -0,0 +1,114 @@
+name: Build and Test Python Wheels
+
+on:
+  push:
+
+defaults:
+  run:
+    shell: bash --noprofile --norc -euo pipefail {0}
+
+jobs:
+  build-cccl:
+    name: Build CCCL Wheels
+    uses: ./.github/workflows/run-ci-script.yml
+    with:
+      script-command: ci/build_cuda_cccl_python.sh -py-version ${{ matrix.python-version }}
+      python-version: ${{ matrix.python-version }}
+      upload-artifact: true
+      upload-artifact-name: wheel-cccl-${{ matrix.python-version }}
+      upload-artifact-path: ${{github.event.repository.name}}/wheelhouse/
+      runner: linux-amd64-cpu16
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12', '3.13']
+    permissions:
+      id-token: write
+      contents: read
+
+  test-cccl:
+    name: Test CCCL Wheels
+    needs: build-cccl
+    uses: ./.github/workflows/run-ci-script.yml
+    with:
+      script-command: ci/test_cuda_cccl_python.sh -py-version ${{ matrix.python-version }}
+      python-version: ${{ matrix.python-version }}
+      needs-artifact: true
+      artifact-name: wheel-cccl-${{ matrix.python-version }}
+      artifact-path: ${{github.event.repository.name}}/wheelhouse/
+      runner: linux-amd64-cpu16
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12', '3.13']
+    permissions:
+      id-token: write
+      contents: read
+
+  build-cooperative:
+    name: Build Cooperative Wheels
+    uses: ./.github/workflows/run-ci-script.yml
+    with:
+      script-command: ci/build_cuda_cooperative_python.sh -py-version ${{ matrix.python-version }}
+      python-version: ${{ matrix.python-version }}
+      upload-artifact: true
+      upload-artifact-name: wheel-cooperative-${{ matrix.python-version }}
+      upload-artifact-path: ${{github.event.repository.name}}/wheelhouse/
+      runner: linux-amd64-cpu16
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12', '3.13']
+    permissions:
+      id-token: write
+      contents: read
+
+  test-cooperative:
+    name: Test Cooperative Wheels
+    needs: build-cooperative
+    uses: ./.github/workflows/run-ci-script.yml
+    with:
+      script-command: ci/test_cuda_cooperative_python.sh -py-version ${{ matrix.python-version }}
+      python-version: ${{ matrix.python-version }}
+      needs-artifact: true
+      artifact-name: wheel-cooperative-${{ matrix.python-version }}
+      artifact-path: ${{github.event.repository.name}}/wheelhouse/
+      runner: linux-amd64-gpu-rtxa6000-latest-1
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12', '3.13']
+    permissions:
+      id-token: write
+      contents: read
+
+  build-parallel:
+    name: Build Parallel Wheels
```

xref https://github.com/NVIDIA/cccl/pull/4679#discussion_r2093688121
```suggestion
  build-parallel:
    name: Build Parallel Wheels
    needs: build-cccl
```

---

Thinking about this for a few more seconds, perhaps we only need the `cuda-cccl` wheel at test time, not build time?

---

I mean we still build it in the same CI runs, just different stages. Repeatedly building the same wheel in different CI jobs seems odd to me if we can make it safe by declaring job dependencies:
```
build-parallel  <---  test-parallel
                            |
      -----------------------
      |
      v
build-cccl  <---  test-cccl
      ^
      |
      ------------------
                       |
build-coop  <---  test-coop
```

---

Yes, that's the right dependency tree

---

If it's alright, I'm going to tackle this in a follow-up PR. It's going to involve changes to the `upload/download-artifacts` steps to work with multiple artifacts.

There are some examples of how to do that [here](https://github.com/actions/download-artifact?tab=readme-ov-file#download-multiple-filtered-artifacts-to-the-same-directory). But it's non-trivial enough that I think it warrants a separate PR.

---

(xref https://github.com/NVIDIA/cccl/issues/4767)

---

cuda-python repo can be followed too